src/transifex_api/__init__.py

```diff
@@ -75,6 +77,39 @@ def upload(cls, resource, content, interval=5):
                                       files={'content': content})
 
         while True:
+            if upload.get('errors') and len(upload.errors) > 0:
+                raise JsonApiException(409, upload.errors)
+            if upload.redirect:
+                return upload.follow()
+            time.sleep(interval)
+            upload.reload()
+
+
+@_api.register
+class ResourceTranslationsAsyncUpload(Resource):
+    TYPE = "resource_translations_async_uploads"
+
+    @classmethod
+    def upload(cls, resource, content, language, interval=5, file_type='default'):
+        """ Upload translation content with multipart/form-data.
+
+            :param resource: A (transifex) Resource instance or ID
+            :param content: A string or file-like object
+            :param language: A (transifex) Language instance or ID
+            :param interval: How often (in seconds) to poll for the completion
+                             of the upload job
+            :param file_type: The content file type
+        """
+
+        if isinstance(resource, Resource):
+            resource = resource.id
+
+        upload = cls.create_with_form(data={'resource': resource, 'language': language, 'file_type': file_type},
+                                      files={'content': content})
+
+        while True:
+            if upload.get('errors') and len(upload.errors) > 0:
+                raise JsonApiException(409, upload.errors)
```

Comment:
The initializer to `JsonApiException` will not like these arguments, I think:

- https://github.com/kbairak/transifex-api-python/blob/master/src/jsonapi/exceptions.py#L38
- https://github.com/kbairak/transifex-api-python/blob/master/src/jsonapi/exceptions.py#L61

I believe a simple `ValueError(*upload.errors)` would be better here.

Reply:
You're right, the code caused me great headache (hidden bug). Couldn't make also `ValueError()` work for me and resorted to building a dictionary compatible to`JsonApiError`.